permissions/index.js

```diff
@@ -1,40 +1,75 @@
 const User = require('./user');
 const { SafeRouter, publicEndpoint } = require('./safe-router');
-const { applyPermissions: applyZmPermissions } = require('./rules/zm');
-const { applyPermissions: applyZaPermissions } = require('./rules/za');
-const { applyPermissions: applyEtPermissions } = require('./rules/et');
-const { applyPermissions: applyMwPermissions } = require('./rules/mw');
-const { applyPermissions: applyBrPermissions } = require('./rules/br');
-const { applyPermissions: applyInPermissions } = require('./rules/in');
-const { applyPermissions: applyJmPermissions } = require('./rules/jm');
-const { applyPermissions: applyCaPermissions } = require('./rules/ca');
-const { applyPermissions: applyOpenPermissions } = require('./rules/open');
+const { setupCanForRules } = require('./setupCanForRules');
+
+const openRules = require('./rules/open.json');
+const brRules = require('./rules/br.json');
+const caRules = require('./rules/ca.json');
+const etRules = require('./rules/et.json');
+const inRules = require('./rules/in.json');
+const jmRules = require('./rules/jm.json');
+const mwRules = require('./rules/mw.json');
+const zaRules = require('./rules/za.json');
+const zmRules = require('./rules/zm.json');
+
 const { canEditCase, canViewPostSurvey } = require('./middlewares');
 
-const applyPermissions = {
-  zm: applyZmPermissions,
-  za: applyZaPermissions,
-  et: applyEtPermissions,
-  mw: applyMwPermissions,
-  br: applyBrPermissions,
-  in: applyInPermissions,
-  jm: applyJmPermissions,
-  ca: applyCaPermissions,
+// TODO: maybe factor out so it's easier to build the API that retrieves this to the frontend
+const rulesMap = {
+  br: brRules,
+  ca: caRules,
+  et: etRules,
+  in: inRules,
+  jm: jmRules,
+  mw: mwRules,
+  za: zaRules,
+  zm: zmRules,
+};
+
+/**
+ * @type {{ [key in keyof typeof rulesMap]: ReturnType<typeof setupCanForRules> }}
+ */
+const initializedCanMap = Object.entries(rulesMap).reduce((accum, [key, rules]) => {
+  const can = setupCanForRules(rules);
+  return {
+    ...accum,
+    [key]: can,
+  };
+}, {});
+
+const initializedCanOpenRules = setupCanForRules(openRules);
+
+/**
+ * Applies the permissions if valid.
+ * @param {import('express').Request} req
+ * @param {ReturnType<typeof setupCanForRules>} initializedCan
+ * @param {string} permissionsConfig
+ * @throws Will throw if initializedCan is not a function
+ */
+const applyPermissions = (req, initializedCan, permissionsConfig) => {
+  if (!initializedCan) throw new Error(`Cannot find rules for ${permissionsConfig}`);
+
+  if (typeof initializedCan === 'string')
+    throw new Error(`Error in rules for ${permissionsConfig}. Error: ${initializedCan}`);
+
+  if (typeof initializedCan !== 'function')
+    throw new Error(`Error in rules for ${permissionsConfig}. Error: can is not a function.`);
+
+  req.can = initializedCan;
 };
 
 const setupPermissions = (req, res, next) => {
   if (process.env.NODE_ENV === 'test') {
```

Comment:
We should be looking at moving away from checking NODE_ENV in the plugin, I think it's partially deprecated and driving functionality directly off environment isn't best practice anyway - but that's for another PR

Reply:
What would be a better approach for this use case? We could set the "open permissions" to be used in the same way as the other ones, but that's also provided as env vars :/

Reply:
Env vars are fine, it's just driving things off NODE_ENV, rather than specific variables for individual settings (e.g. DISABLE_PERMISSIONS in this case) which isn't great practice, because it locks you into a bunch of hardcoded configuration flags based on the environment you pick, rather than being able to set them individually.

With individual settings you can still provide a set of defaults per environment, but they can be changed and overridden, unlike if settings are determined directly by the environment setting